### PR TITLE
Seqr loader to MultiCohorts

### PR DESCRIPTION
**Pull request by MattWellie**
n.b. a test is failing here - that is addressed in #865, but requires a slight restructure and should be evaluated separately

- Changes everything after Genotype and Align to use MultiCohorts
- In general removes a lot of repeat calls to expected_outputs. Instead of:

```
output1 = self.expected_outputs(multicohort)['foo']
output2 = self.expected_outputs(multicohort)['bar']
output3 = self.expected_outputs(multicohort)['baz']
```

I've replaced this with one call to get the dict, and pulling values from it instead:
```
outputs = self.expected_outputs(multicohort)
output1 = outputs['foo']
output2 = outputs['bar']
output3 = outputs['baz']
```

- In general the convention in these Stages of the expected_outputs containing an output path as a `Path` (so that the pipeline will check for presence/absence), and passing a tmp path as a String (so the pipeline won't check it) has been removed - there's no need for tmp_path to be in the output dict, it's created as needed using `self.tmp_prefix` (a property of the Stage itself, not exclusive to the expected_outputs method)

---

JointGenotyping
- Simple, one Stage, turns it into MultiCohorts
- Uses MultiCohorts.get_sequencing_groups instead

VQSR
- Simple(ish) 1-stage workflow
- number_of_sgids is only generated/counted once
- moves from Cohort -> MultiCohort
- The tmp_path here is actually not in tmp. I've not changed the convention here but *maybe we should*

VEP
- Simple(ish) 1-stage workflow
- number_of_sgids is only generated/counted once
- moves from Cohort -> MultiCohort
- The tmp_path here is actually not in tmp. I've not changed the convention here but *maybe we should*

seqr_loader
- Removes some update_analysis_meta callables. I'm ambivalent about this, but I don't think these currently work, and I don't think they're needed
- AnnotateCohort -> MultiCohort
- AnnotateDataset remains as Dataset, but picks up results from `dataset.cohort.multicohort` instead of `dataset.cohort`
- Removes `_sg_vcf_meta` which is imported in (and not really required in) HappyValidation

happy_validation
- Remove the `_sg_vcf_meta` update_analysis_meta callable

### CODE REVIEW

File: cpg_workflows/stages/seqr_loader.py [hunk 1]
```diff
@@ -150,26 +93,19 @@ def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput | None
         assert dataset.cohort
         mt_path = inputs.as_path(target=dataset.cohort, stage=AnnotateCohort, key='mt')
```
Comment (EddieLF) [inline cpg_workflows/stages/seqr_loader.py:94]:
```suggestion
        assert dataset.cohort.multicohort
        mt_path = inputs.as_path(target=dataset.cohort.multicohort, stage=AnnotateCohort, key='mt')
```

**Review by EddieLF — APPROVED** [review]
Changes look good to me, cheers @MattWellie.

File: cpg_workflows/stages/joint_genotyping.py [hunk 2]
```diff
@@ -59,19 +50,19 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
...
         ]
 
         jc_jobs = joint_genotyping.make_joint_genotyping_jobs(
             b=get_batch(),
-            out_vcf_path=vcf_path,
-            out_siteonly_vcf_path=siteonly_vcf_path,
-            tmp_bucket=to_path(self.expected_outputs(cohort)['tmp_prefix']),
+            out_vcf_path=outputs['vcf'],
+            out_siteonly_vcf_path=outputs['siteonly'],
+            tmp_bucket=self.prefix / 'tmp',
```
Comment (cassimons) [inline cpg_workflows/stages/joint_genotyping.py:65]: Should we add a comment here why it is not going to the temp bucket?

**Review by cassimons — APPROVED** [review]
Thanks Matt. Looks good and I like the clean up!

File: cpg_workflows/stages/joint_genotyping.py (hunk 2 above)
Comment (MattWellie) [reply cpg_workflows/stages/joint_genotyping.py:65]: Ugh, I've made this into a real TMP path - we should review the stuff that's already being retained (e.g. [here](https://console.cloud.google.com/storage/browser/cpg-seqr-main/seqr_loader/6c662c33c550a9c69d55c909c1388cc7e4930b_3755/JointGenotyping/tmp)) - looks like we're keeping multiple terabytes of additional data per joint call that should probably have been in tmp

**Pull request merged**